sbi/inference/snpe/snpe_c.py

```diff
@@ -152,13 +180,123 @@ def __call__(
         self._num_atoms = num_atoms
         kwargs = del_entries(locals(), entries=("self", "__class__", "num_atoms"))
 
+        if proposal is not None:
+            self.use_non_atomic_loss = (
+                isinstance(proposal.net._distribution, mdn)
+                and isinstance(self._posterior.net._distribution, mdn)
+                and (
+                    isinstance(self._prior, utils.BoxUniform)
+                    or isinstance(self._prior, mvn.MultivariateNormal)
+                )
+            )
+
+            algorithm = "non-atomic" if self.use_non_atomic_loss else "atomic"
+            print(f"Using SNPE-C with {algorithm} loss")
```

Comment:
fall back onto logging module

Reply:
I decided to not go for logging here. The reason is that the default level is warning. I do not want to make this a warning, but rather an info. However, the user will not see it then unless they manually change logging level.